lib/Doctrine/DBAL/Driver/Mysqli/MysqliStatement.php

```diff
@@ -272,10 +274,17 @@ private function sendLongData($streams)
     private function bindUntypedValues(array $values)
     {
         $params = [];
-        $types  = str_repeat('s', count($values));
-
+        $types  = '';
         foreach ($values as &$v) {
             $params[] =& $v;
+            // fix for issue #3631 - detect parameter types as they have to be bound differently
```

Comment:
Why would we want to autodetect the type of untyped parameters? They should be bound as strings by design.

Reply:
@morozov,

Removed this for now (was missing the case for double / d anyway, as I was still testing things out) - this is for a separate "part II" of the bug where another test fails when under de_DE when dealing with floats.

I may add it back later, or do a follow on PR once this is accepted.